# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/loader/NetworkLoadHandler.cpp
# Commit: 6118d850a96d
# Full Hash: 6118d850a96dfa307d91cb73392c659ffb328335
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-18 15:46:32
# Regressor Bug: 1800496
# File Overlap Count: 1
# Description:
#   Bug 1800496 - Re-introduce channel tracking for cancellation; r=asuth
#   
#   This was removed as part of an attempt to simplify worker cancellaton. However, this channel cancellation is important, as without it the timing changes when we finish a request. Without it, we continue to load a script, even after a worker has been shut down.
#   
#   This tracking will help us reimplement cancellation.
# ==============================================================================

diff -r eafec005c1bf -r 6118d850a96d dom/workers/loader/NetworkLoadHandler.cpp
--- a/dom/workers/loader/NetworkLoadHandler.cpp	Fri Nov 18 10:02:26 2022 +0000
+++ b/dom/workers/loader/NetworkLoadHandler.cpp	Fri Nov 18 10:02:26 2022 +0000
@@ -75,6 +75,12 @@
   MOZ_ASSERT(!mRequestHandle->IsEmpty());
   WorkerLoadContext* loadContext = mRequestHandle->GetContext();
 
+  if (!loadContext->mChannel) {
+    return NS_BINDING_ABORTED;
+  }
+
+  loadContext->mChannel = nullptr;
+
   if (NS_FAILED(aStatus)) {
     return aStatus;
   }